File: rust/leetcode/src/q2/q2_minimumDeletions.cpp
#include <string>
#include <algorithm>
class Solution {
public:
    int minimumDeletions(const std::string& s) {
        int lb = 0, ra = std::count(s.begin(), s.end(), 'a');
        int ans = ra;
        for (char c : s) {
            ra -= c == 'a';
            ans = std::min(ans, lb + ra);
            lb += c == 'b';
        }
        return ans;
    }
};

File: rust/leetcode/src/q2/q2_minimum_deletions.rs
impl Solution {
    pub fn minimum_deletions(nums: Vec<i32>) -> i32 {
        let ((mut aa, _), (mut bb, _)) =
            nums.iter()
                .enumerate()
                .fold(((0, nums[0]), (0, nums[0])), |(aa, bb), (cc, &dd)| {
                    (
                        if dd < aa.1 { (cc, dd) } else { aa },
                        if dd > bb.1 { (cc, dd) } else { bb },
                    )
                });

        if aa > bb {
            std::mem::swap(&mut aa, &mut bb);
        }

        let cc = nums.len();
        (bb + 1).min(cc - aa).min(aa + 1 + cc - bb) as i32
    }
}
